### ISSUE TITLE
error: wrong number of arguments (2 for 1)

### ISSUE THREAD

**Issue by ginkel** [jedcn/reveal-ck#58]
When running

    reveal-ck generate
    reveal-ck serve

on a trivial `slides.md`

    ## Test

under ruby 2.1.5 I get the following error output:

    D, [2015-07-16T07:29:45.184896 #28278] DEBUG -- : Celluloid 0.17.0 is running in BACKPORTED mode. [ http://git.io/vJf3J ]
     ------------------------------------------------------------
    [ reveal-ck ] Serving up slide content in 'slides/'.
    [ reveal-ck ] Open your browser to 'http://localhost:10000'.
    [ reveal-ck ] Press CTRL-C to stop.
    ------------------------------------------------------------
    [ reveal-ck ] Getting Ready to Reload Browsers.
    [ reveal-ck ] Getting Ready to Rebuild Slides.
    error: wrong number of arguments (2 for 1)

`gem list --local`:

    *** LOCAL GEMS ***
    
    activesupport (4.2.3)
    bigdecimal (1.2.4)
    bundler (1.7.9)
    bundler-unload (1.0.2)
    celluloid (0.17.0)
    celluloid-essentials (0.20.1)
    celluloid-extras (0.20.0)
    celluloid-fsm (0.20.0)
    celluloid-pool (0.20.0)
    celluloid-supervision (0.20.0)
    coderay (1.1.0)
    docile (1.1.5)
    dotenv (2.0.2)
    em-websocket (0.5.1)
    eventmachine (1.0.7)
    executable-hooks (1.3.2)
    ffi (1.9.10)
    formatador (0.2.5)
    gem-wrappers (1.2.7)
    gemoji (2.1.0)
    gli (2.12.3)
    guard (2.12.4)
    guard-livereload (2.4.0)
    haml (4.0.6)
    hitimes (1.2.2)
    html-pipeline (1.11.0)
    http_parser.rb (0.6.0)
    i18n (0.7.0)
    io-console (0.4.2)
    json (1.8.3, 1.8.1)
    listen (2.8.5)
    lumberjack (1.0.9)
    method_source (0.8.2)
    mini_portile (0.6.2)
    minitest (5.7.0, 4.7.5)
    multi_json (1.11.2)
    nenv (0.2.0)
    nokogiri (1.6.6.2)
    notiffany (0.0.6)
    pry (0.10.1)
    psych (2.0.5)
    rack (1.6.0)
    rack-livereload (0.3.15)
    rake (10.4.2, 10.1.0)
    rb-fsevent (0.9.5)
    rb-inotify (0.9.5)
    rdoc (4.1.0)
    redcarpet (3.2.2)
    reveal-ck (3.2.0)
    rinku (1.7.3)
    rspec-logsplit (0.1.3)
    rubygems-bundler (1.4.4)
    rvm (1.11.3.9)
    shellany (0.0.1)
    slim (3.0.2)
    slop (3.6.0)
    temple (0.7.6)
    test-unit (2.1.5.0)
    thor (0.19.1)
    thread_safe (0.3.5)
    tilt (2.0.1)
    timers (4.0.1)
    tzinfo (1.2.2)

**Comment by jedcn** [comment]
Hello, and thanks for logging an issue @ginkel!

This is probably happening because one of reveal-ck's dependencies (celluloid) has released a new version that has unexpected changes.

reveal-ck depends on many gems for dependencies, and it *locks* many of these dependencies down to a specific version in its `.gemspec`. It locks them down to avoid taking on breaking changes when new versions of dependencies are released.

When it comes to celluloid, it appears that version 0.17 was released in the past two weeks, and contains [several breaking changes][celluloid-changes].

[celluloid-changes]: https://github.com/celluloid/celluloid/blob/master/CHANGES.md

What I'm seeing is as follows:

* `reveal-ck generate` works with a trivial slides.md
* `reveal-ck serve` does not work with a trivial slides.md. It exits with the same output you describe.
* Both commands display the Celluloid 0.17 DEBUG message: `Celluloid 0.17.0 is running in BACKPORTED mode.`

Here's what I suggest as a temporary work around:

Create a `Gemfile` with the following content:

```ruby
source 'https://rubygems.org'

gem 'reveal-ck', '3.2.0'
gem 'celluloid', '0.16.0'
```

And then run `bundle` in the same directory as the `Gemfile` to get an install of reveal-ck with celluloid `0.16.0`. This command should complete without error.

Once you've successfully run a bundle,

* `bundle show celluloid` will show the 0.16.0 release of the gem
* `bundle exec reveal-ck generate` will continue to work.
* `bundle exec reveal-ck serve` will work.
* Neither `reveal-ck` command will print out the message: `Celluloid 0.17.0 is running in BACKPORTED mode.`

If this is working, it is because `bundler` is limiting reveal-ck to work with only celluloid 0.16.0.

Can you please confirm that this is working for you? Thanks!

In the near future, I will push a new version of reveal-ck that either fixes this issue or locks `celluloid` to 0.16.0. It looks like celluloid is making a bunch of changes right now, and I'll probably lock the version until things settle down.

**Comment by ginkel** [comment]
Excellent, thanks for the speedy reply! I can confirm that pinning celluloid to v0.16.0 indeed solves the issue.

**Comment by jedcn** [comment]
Thanks for letting me know that workaround worked for you.

I took the summer off, but now I'm back.

I've upgraded to the latest compatible versions of all the gems, and this workaround should no longer be necessary now that I've published version 3.3.0 of reveal-ck.

See https://github.com/jedcn/reveal-ck/releases/tag/v3.3.0 for more info if you'd like it.

Thanks for submitting this issue!!
